internal/app/wwctl/ssh/main.go

```diff
@@ -38,23 +38,36 @@ func CobraRunE(cmd *cobra.Command, args []string) error {
 		cmd.Usage()
 		os.Exit(1)
 	}
-
+	globalPrimary := ""
 	for _, node := range nodes {
-
-		if _, ok := node.NetDevs["default"]; !ok {
-			fmt.Fprintf(os.Stderr, "%s: Default network device doesn't exist\n", node.Id.Get())
+		var primaryNet string
+		if _, ok := node.NetDevs[globalPrimary]; ok {
+			if node.NetDevs[globalPrimary].Primary.GetB() {
+				primaryNet = globalPrimary
+			}
+		}
+		for netName := range node.NetDevs {
```

Comment:
I think this `for` loop needs to be wrapped with an `if` that tests whether `primaryNet` is already set. Otherwise, the discovery from `globalPrimary` above is discarded regardless.

Reply:
Yes, so if as assumed further down, that there is always *one* primary device, I can discard the globalPrimary.